app/src/main/java/org/p2p/wallet/infrastructure/network/data/InstructionErrorType.kt

```diff
@@ -0,0 +1,22 @@
+package org.p2p.wallet.infrastructure.network.data
+
+import com.google.gson.JsonElement
+
+private const val KEY_CUSTOM_INSTRUCTION_ERROR_ID = "Custom"
+
+sealed interface InstructionErrorType {
+    companion object {
+        // can be { "Custom" : 1111 } or just plain "SomeError"
+        fun from(instructionErrorType: JsonElement): InstructionErrorType {
+            val isCustom = instructionErrorType.isJsonPrimitive
```

Comment:
Why if it's primitive we cast it as JsonObject ? if I am not wrong, String is primitive too.

Reply:
thanks for pointing this out, made a test to check this stuff